server/app/api/v1/endpoints/subscriptions.py

```
@@ -0,0 +1,2016 @@
+    result = stripe_service.validate_coupon(
+        code,
+        plan_id=resolved_plan_id,
+        plan_stripe_price_id=plan_stripe_price_id,
+        plan_stripe_product_id=plan_stripe_product_id,
+    )
+    return result
+
+
+@router.post("/portal-session-for-customer")
+async def create_billing_portal_session_for_customer(
+    rooftop_id: Optional[UUID] = Query(None, description="Rooftop ID for rooftop-level billing; omit for org-level"),
```

Ruff (0.14.14)

[warning] Do not perform function call `Query` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)